### ISSUE TITLE
New semantic analyzer: report undefined types in signatures

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#6425]
Currently undefined types in function annotations don't generate an error when using the new semantic analyzer. Example:

```py
def f(x: adsf) -> None:  # No error
    pass
```
